include/util/extract-conditionals.hpp

```diff
@@ -0,0 +1,386 @@
+#include "util/conditional_restrictions.hpp"
```

Comment:
This file does not have an include guard and is not included besides from `src/tools/extract-conditionals.cpp`. It needs include guard and should be moved to `include/tools`.